### ISSUE TITLE
ttnn.sort returns out-of-range indices when the input contains padding sentinel value

### ISSUE THREAD

**Issue by iwroszTT** [tenstorrent/tt-metal#53326]
ttnn.sort pads the sort dimension up to the bitonic network size using a sentinel value: +inf for ascending direction, -inf for descending. When the input itself contains that value, the padding lanes tie with real elements and the sort emits the padding lanes' indices into the logical output region.

The returned values are correct (they match torch.sort exactly), so value-only tests pass, but the indices tensor contains errors.

To Reproduce
                                                                                                                           
 ```
                                                                                                                   
  import pytest                                                                                                                             
  import torch                                                                                                                              
  import ttnn                                                                                                                               
                                                                                                                      
  @pytest.mark.parametrize(                                                                                                                 
      "n, descending, fill",                                                                                                                
      [                                                                                                                                     
          (6144, True, float("-inf")),   # 1948 out-of-range indices, max index 8191
          (151936, True, float("-inf")), # #26057 shape: max index 262143 = 2**18 - 1
          (151936, False, float("inf")), # ascending is symmetric: 109916 out-of-range
      ],
  )
  def test_sort_pad_index_leak(n, descending, fill, device):
      torch.manual_seed(0)
      input = torch.full((1, n), fill, dtype=torch.float32)
      input[..., torch.randperm(n)[:328]] = torch.randn(328) * 8.0

      ttnn_input = ttnn.from_torch(input, ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
      _, ttnn_indices = ttnn.sort(ttnn_input, descending=descending)   

      indices = ttnn.to_torch(ttnn_indices).reshape(-1)[:n].long()
      assert indices.max() < n  # FAILS: indices point into the pow2 padding region
```

Expected behavior                                                    
                                                                       
  Indices are always in [0, N) regardless of input content.

Observed behavior (Blackhole p100a, main @ e16f6cbfb09, 2026-08-17, deterministic)
                                                                       
  ```
┌──────────────────────────┬────────────┬────────────┬──────────────────────┬─────────────────────┐
  │          shape           │ descending │ input fill │ out-of-range indices │      max index      │
  ├──────────────────────────┼────────────┼────────────┼──────────────────────┼─────────────────────┤
  │ [1, 6144]                │ True       │ -inf       │ 1948                 │ 8191                │
  ├──────────────────────────┼────────────┼────────────┼──────────────────────┼─────────────────────┤
  │ [1, 8192] (pow2 control) │ True       │ -inf       │ 0                    │ 8191 (= n-1, valid) │
  ├──────────────────────────┼────────────┼────────────┼──────────────────────┼─────────────────────┤
  │ [1, 151936]              │ True       │ -inf       │ (fails test)         │ 262143              │
  ├──────────────────────────┼────────────┼────────────┼──────────────────────┼─────────────────────┤
  │ [1, 151936]              │ False      │ +inf       │ 109916               │ 262143              │
  └──────────────────────────┴────────────┴────────────┴──────────────────────┴─────────────────────┘

```

Notes on root cause / fix direction

  - Sentinel selection is applied via fill_implicit_tile_padding and ttnn::pad
  - No sentinel can sort strictly after a value range that includes ±inf (or the uint16 extremes), so this is not fixable by picking a better pad constant.
  - The composite (value, original_index) tie-break comparator planned for #33492 would likely fix this as a side effect: padding indices (>= N) are larger than all logical indices.

  Related issues
  - Found while verifying #26057
  - Note that #43917 proposes padding non-pow2 widths with ±inf sentinels to unlock the fast multicore path. That approach will suffer from this bug.
